src/content/docs/en/reference/image-service-reference.mdx

```diff
@@ -0,0 +1,113 @@
+---
+title: Image Service API
+i18nReady: false
+---
+
+Astro [assets](/en/guides/assets/) were designed to make it easy for any image optimization service to build a service on top of Astro.
+
+:::caution
+The Image Services API is an _experimental_ Astro feature introduced in v2.1. This API is subject to change before it is marked as stable.
+:::
+
+## What is an Image Service?
+
+Astro provides two types of image services: Local and External.
+
+- **Local services** handle image transformations directly at build for static sites, or at runtime in dev / SSR. These are often wrappers around libraries like Sharp, ImageMagick or Squoosh.
+- **External services** point to URLs and can add support for services such as Cloudinary, Vercel or any RIAPI-compliant server.
+
+## Building using the Image Services API
+
+Services definitions take the shape of an exported default object with various required methods ("hooks") to create the required properties for your image. 
+
+Local services have a `transform` method available to perform transformations on your image. External services can only return the final URL and image attributes.
+
+### Local Service
+
+```js
+import type { LocalImageService } from "astro";
+
+const service: LocalImageService = {
+  getURL(options: ImageTransform) {
+    return `/my_super_endpoint_that_transforms_images?w=${options.width}`;
+  },
+  parseURL(url: URL) {
+    return {
+      width: url.searchParams.get("w"),
+    };
+  },
+  transform(options: ImageTransform) {
+    const { buffer } = mySuperLibraryThatEncodesImages(options);
+
+    return {
+      data: buffer,
+      format: options.format,
+    };
+  },
+};
+
+export default service;
+```
+
+### External Service
+
+```js
+import type { ExternalImageService } from "astro";
+
+const service: ExternalImageService = {
+  getURL(options: ImageTransform) {
+    return `https://mysupercdn.com/${options.src}?q=${options.quality}`;
+  },
+};
+
+
+export default service;
+```
+
+## Hooks
+
+### `getURL()`
+
+**Required for local and external services**
+
+`getURL(options: ImageTransform): string`
```

Comment:
Here I am fine with leaving it as `ImageTransform` since it is explained below what `options` contain.